### DISCUSSION TITLE
cat not printing spaces while printing /proc/pid/cmdline

### DISCUSSION THREAD

**Discussion by techneophyte** [bpftrace/bpftrace#3455] [Q&A]
The cat function is not printing spaces when I try to print `/proc/pid/cmdline`. This is because cmdline stores spaces as null chars and cat is not converting them back to space when used. 

 I found an alternative to this to use `tr` like `cat /proc/pid/cmdline | tr '\0' ' '`. But when I use this in bpftrace cat function like below, I do not see any output.
` cat("/proc/%d/cmdline | tr '\0' ' '", pid);`
Any suggestions on how to reslove this?

**Comment by ajor** [comment]
For security, the `cat` builtin function can only print out file contents and nothing else.

To run arbitrary commands, you'll need to use the `system` function with `--unsafe`:
```
system("cat /proc/%d/cmdline | tr '\0' ' '", pid)
```

Hopefully this is a start, but I haven't worked out how to make the null byte work yet.
